plugins/de.cognicrypt.staticanalyzer/src/de/cognicrypt/staticanalyzer/markerresolution/QuickFixer.java

```diff
@@ -13,17 +15,33 @@
  * @author André Sonntag
  */
 public class QuickFixer implements IMarkerResolutionGenerator {
+	
+	ArrayList<IMarkerResolution> quickFixes;
+
 	@Override
 	public IMarkerResolution[] getResolutions(final IMarker mk) {
-
+		quickFixes = new ArrayList<>();
 		String message = "";
+		String errorType = "";
+		
 		try {
+			errorType = (String) mk.getAttribute("errorType");
 			message = (String) mk.getAttribute(IMarker.MESSAGE);
+			quickFixes.add(new SuppressWarningFix(Constants.SUPPRESSWARNING_FIX + message));
+
+			//TODO: currently the ensuresPredicate quickfix is disabled.
+//			if (errorType.equals(Constants.REQUIRED_PREDICATE_MARKER_TYPE)) {
+//				String predicate = (String) mk.getAttribute("predicate");
+//				
+//				if(predicate.equals("generatedKey") || predicate.equals("randomized")) {
+//					quickFixes.add(new EnsuresPredicateFix("This object comes from a stream/database/other external source and is actually secure."));
+//				}

```

Comment:
Is this code needed? If not, please remove.

Reply:
This is the code for enabling the quickfix

Reply:
Why is it commented then?